### PR TITLE
Pmanoj/falcon component changes

### PR DESCRIPTION
**Pull request by jpmann**
Updates:
======
1. Update the lora config to include target modules for falcon model family
2. Updated the falcon family to support QLoRA
3. Hack for AutoModelForSequenceClassification, AutoModelForTokenClassification, AutoModelForQuestionAnswering class method from_pretrained to add trust_remote_code=True

Sanity runs - 
=======
1. text classification - [Link](https://ml.azure.com/experiments/id/ace89a6d-e8c4-4a6b-b880-441e68b11e33/runs/0edcf441-877b-4206-adb4-61fd24c8da1e?wsid=/subscriptions/9ec1d932-0f3f-486c-acc6-e7d78b358f9b/resourceGroups/AML_shared/providers/Microsoft.MachineLearningServices/workspaces/AML_shared_eus_ws&tid=72f988bf-86f1-41af-91ab-2d7cd011db47&pipelineChildMetrics=true)
4. QnA - [Link](https://ml.azure.com/experiments/id/ace89a6d-e8c4-4a6b-b880-441e68b11e33/runs/f9609ec6-1a44-4a36-9b14-a95faa833a15?wsid=/subscriptions/9ec1d932-0f3f-486c-acc6-e7d78b358f9b/resourceGroups/AML_shared/providers/Microsoft.MachineLearningServices/workspaces/AML_shared_eus_ws&tid=72f988bf-86f1-41af-91ab-2d7cd011db47)

### CODE REVIEW

File: assets/training/finetune_acft_hf_nlp/src/finetune/finetune.py [hunk 1]
```diff
@@ -617,6 +657,47 @@ def main():
...
+            LoggingLiterals.PROJECT_NAME: PROJECT_NAME,
+            LoggingLiterals.PROJECT_VERSION_NUMBER: VERSION,
+            LoggingLiterals.COMPONENT_NAME: COMPONENT_NAME
+        },
+        azureml_pkg_denylist_logging_patterns=LOGS_TO_BE_FILTERED_IN_APPINSIGHTS,
+    )
+
+    # Updata lora target modules for falcon
+    update_lora_target_modules()
+    if hasattr(args, "model_type") and args.model_type in [
```
Comment (amangupta26) [inline assets/training/finetune_acft_hf_nlp/src/finetune/finetune.py:676]: Add comment on why this was needed?

File: assets/training/finetune_acft_hf_nlp/src/finetune/finetune.py [hunk 2]
```diff
@@ -617,6 +657,47 @@ def main():
...
+        acft_custom_dimensions={
+            LoggingLiterals.PROJECT_NAME: PROJECT_NAME,
+            LoggingLiterals.PROJECT_VERSION_NUMBER: VERSION,
+            LoggingLiterals.COMPONENT_NAME: COMPONENT_NAME
+        },
+        azureml_pkg_denylist_logging_patterns=LOGS_TO_BE_FILTERED_IN_APPINSIGHTS,
+    )
+
+    # Updata lora target modules for falcon
+    update_lora_target_modules()
```
Comment (amangupta26) [inline assets/training/finetune_acft_hf_nlp/src/finetune/finetune.py:675]: Can we do this only if the model is Falcon?

File: assets/training/finetune_acft_hf_nlp/src/finetune/finetune.py (hunk 1 above)
Comment (jpmann) [reply assets/training/finetune_acft_hf_nlp/src/finetune/finetune.py:678]: added comment

File: assets/training/finetune_acft_hf_nlp/src/finetune/finetune.py (hunk 2 above)
Comment (jpmann) [reply assets/training/finetune_acft_hf_nlp/src/finetune/finetune.py]: done

**Review by aggarwal-k — APPROVED** [review]

**Review by amangupta26 — APPROVED** [review]

**Pull request merged**